### PR TITLE
Toggle primary read-only when disk capacity hits threshold

### PR DESCRIPTION
**Pull request by davissp14**
This PR ensures the primary is made read-only when disk capacity exceeds 90%.  

I've introduced a new PG health check called `disk-capacity`.  This health check has quite a bit of overlap with the existing VM check communicating disk usage, but given the implications and message I feel it should be separated.   

### How it works

On an interval defined by the health check, we will calculate disk usage and evaluate whether capacity has exceeded our pre-defined threshold of 90% ( should be made configurable in the future ). 

When capacity exceeds the pre-defined threshold, we will set a `readonly.lock` file on the filesystem and establish a connection to each user-defined table ( anything that's not Postgres or Repmgr) and make it readonly.  When disk usage falls below the defined threshold, either through file cleanup or volume extension, we will work to turn all tables back to read/write and then clear the `readonly.lock` file on success. 

The `readonly.lock` file is used for two things:

1. If the `readonly.lock` file is present, we know the database(s) have already been made readonly so there's no need to needlessly establish any more connections.

2. It allows us to coordinate intentions across the codebase.  

### Limitations 

Read-only mode is currently enabled at runtime and will be cleared on restart.  That being said, there will be a small window on boot where the primary will accept writes before it is reconfigured back to read-only.   If this becomes a problem, we can look into addressing this. 

https://github.com/fly-apps/postgres-flex/issues/56

### CODE REVIEW

File: internal/flycheck/pg.go [hunk 1]
```diff
@@ -23,18 +25,65 @@ func CheckPostgreSQL(ctx context.Context, checks *check.CheckSuite) (*check.Chec
...
+
+	usedPercentage := float64(size-available) / float64(size) * 100
+
+	// Turn primary read-only
+	if usedPercentage > diskCapacityPercentageThreshold {
+		if err := flypg.SetReadOnly(ctx, node, localConn); err != nil {
+			return "", fmt.Errorf("failed to turn primary readonly: %s", err)
+		}
+
+		return "", fmt.Errorf("%0.1f%% - extend your volume to re-enable writes", usedPercentage)
```
Comment (davissp14) [inline internal/flycheck/pg.go:74]: Not exactly sure why, but i'm getting `MISSING` here.
```
[✗] disk-capacity: 93.2%!e(MISSING)xtend your volume to re-enable writes (2.92ms)
```

**Comment by davissp14** [comment]
https://github.com/fly-apps/postgres-flex/issues/56

File: internal/flycheck/pg.go (hunk 1 above)
Comment (davissp14) [reply internal/flycheck/pg.go:74]: This has been fixed with:

File: internal/flycheck/pg.go [hunk 2]
```diff
@@ -10,9 +10,11 @@ import (
 	"github.com/superfly/fly-checks/check"
 )
 
+// Primary will be made read-only when disk capacity reaches this percentage.
+const diskCapacityPercentageThreshold = 90.0
+
```
Comment (DAlperin) [inline internal/flycheck/pg.go:15]: I've been meaning to make health-check stuff configurable but that can be out of scope of this PR

File: internal/flycheck/pg.go [hunk 3]
```diff
@@ -23,18 +25,65 @@ func CheckPostgreSQL(ctx context.Context, checks *check.CheckSuite) (*check.Chec
...
 	checks.OnCompletion = func() {
 		localConn.Close(ctx)
+		repConn.Close(ctx)
 	}
 
 	checks.AddCheck("connections", func() (string, error) {
 		return connectionCount(ctx, localConn)
 	})
 
+	if member.Role == flypg.PrimaryRoleName && member.Active {
```
[the remark this answers is not in the document]
Comment (DAlperin) [reply internal/flycheck/pg.go:48]: I feel like it might be useful info

[the remark this answers is not in the document]
Comment (davissp14) [reply internal/flycheck/pg.go:48]: There's a VM check that communicates general capacity that should cover that.  It makes me think though that maybe we need a new name for the check.

**Pull request merged**